src/dataframe/dataframe.jl

```diff
@@ -205,6 +208,11 @@ struct DataFrame <: AbstractDataFrame
             end
         end
 
+        for (i, col) in enumerate(columns)
+            fic = firstindex(col)
+            fic != 1 && _onebased_check_error(i, fic)
```

Comment:
Given that this pattern happens in lots of places and that we don't care about performance when an error happens, I would suggest changing `_onebased_check_error` to take `col`:
```suggestion
            firstindex(col) != 1 && _onebased_check_error(i, col)
```